pkg/windows/calico.go

```diff
@@ -60,25 +62,47 @@ func (c *Calico) Setup(ctx context.Context, dataDir string, nodeConfig *daemonco
 		return nil, err
 	}
 
-	secrets, err := client.CoreV1().Secrets(CalicoSystemNamespace).List(ctx, metav1.ListOptions{})
+	sa, err := serviceaccount.GetOrCreateServiceAccount(client.CoreV1(), CalicoSystemNamespace, calicoNode)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, secret := range secrets.Items {
-		if !isCalicoNodeToken(&secret) {
-			continue
-		}
-		value, ok := secret.Data["token"]
-		if !ok {
-			continue
-		}
-		calicoKubeConfig.Token = string(value)
+	t := true
+	sa.AutomountServiceAccountToken = &t
+
+	saClient := client.CoreV1().ServiceAccounts(CalicoSystemNamespace)
+
+	req := authv1.TokenRequest{
```

Comment:
I think you need to also set `audiences` and `expirationSeconds` here? `audiences` should be one or more of the values from https://github.com/k3s-io/k3s/blob/master/pkg/daemons/control/server.go#L191